plugins/json-adapter/src/main/java/org/polypheny/db/adapter/json/JsonSource.java

```diff
@@ -0,0 +1,292 @@
+                } catch ( MalformedURLException e ) {
+                    throw new GenericRuntimeException( e );
+                }
+            }
+            case URL -> {
+                String files = settings.get( "url" );
+                try {
+                    yield new URL( files );
+                } catch ( MalformedURLException e ) {
+                    throw new GenericRuntimeException( e );
+                }
+            }
+        };
+    }
+
+
+    @Override
+    public void updateNamespace( String name, long id ) {
+        namespace = new JsonNamespace( name, id, adapterId );
+    }
+
+
+    @Override
+    public Namespace getCurrentNamespace() {
+        return namespace;
+    }
+
+
+    @Override
+    public void shutdown() {
+        removeInformationPage();
+    }
+
+
+    @Override
+    public List<ExportedDocument> getExportedCollections() {
+        try {
+            return JsonMetaRetriever.getDocuments( jsonFiles );
+        } catch ( IOException e ) {
+            throw new GenericRuntimeException( "Failed to retrieve documents from json file." );
+        }
+    }
+
+
+    @Override
+    public AdapterCatalog getCatalog() {
+        return adapterCatalog;
+    }
+
+
+    @Override
+    public void restoreCollection( AllocationCollection allocation, List<PhysicalEntity> entities, Context context ) {
+        PhysicalEntity collection = entities.get( 0 ); // TODO: set breakpoint and take a look at what's in this list...
+        updateNamespace( collection.getNamespaceName(), collection.getNamespaceId() );
+        try {
+            PhysicalCollection physicalCollection = new JsonCollection( JsonMetaRetriever.findDocumentUrl( jsonFiles, collection.getName() ), collection, allocation.getId(), namespace, this );
+            adapterCatalog.addPhysical( allocation, physicalCollection );
+        } catch ( MalformedURLException | NoSuchFileException e ) {
+            throw new GenericRuntimeException( e );
+        }
+    }
+
+
+    @Override
+    public List<PhysicalEntity> createCollection( Context context, LogicalCollection logical, AllocationCollection allocation ) {
+        PhysicalCollection collection = adapterCatalog.createCollection(
+                logical.getNamespaceName(),
+                logical.getName(),
+                logical,
+                allocation
+        );
+        try {
+            PhysicalCollection physicalCollection = new JsonCollection( JsonMetaRetriever.findDocumentUrl( jsonFiles, collection.getName() ), collection, allocation.getId(), namespace, this );
+            adapterCatalog.replacePhysical( physicalCollection );
+            return List.of( physicalCollection );
+        } catch ( MalformedURLException | NoSuchFileException e ) {
+            throw new GenericRuntimeException( e );
+        }
+    }
+
+
+    @Override
+    public void dropCollection( Context context, AllocationCollection allocation ) {
+        // TODO: What is this supposed to do?
+    }
```

Comment:
Similar to comment to renameLogicalColumn, stores and sources keep track in their physical catalog which entities they actually create/drop (stores) or map (sources). Therefore they have to remove this information if they entity is droped (stores)/un-mapped(sources).